crates/iceberg-catalog/Cargo.toml

```diff
@@ -83,4 +83,3 @@ rand = "0.8.5"
 http-body-util = { workspace = true }
 tower = { workspace = true }
 needs_env_var = { workspace = true }
-tracing-subscriber = "0.3.18"
```

Comment:
i kept this in since it is nice to be able to activate tracing within tests without having to go back to Cargo.toml every time